From af39d2bc8fbab386eb8670030cb80ac6c6ff7ea4 Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Mon, 18 Jan 2016 02:25:57 +0000
Subject: [PATCH 3/4] arm: mach-axxia: kexec_reinit to set L3 CACHE shutdown

Shut down to flush L3 cache for KEXEC to avoid the hung issue for
capture kernel booting, since for ARM capture kernel just one CPU
allowed to boot, and no L3 cache coherency issue existed for SMP
situation, so just shut down the flush to L3 CACHE.

The function calling relationship like below:
kernel_kexec
	machine_kexec
	    kexec_reinit
		flush_l3

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 arch/arm/mach-axxia/axxia.c |    6 ++----
 1 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-axxia/axxia.c b/arch/arm/mach-axxia/axxia.c
index f2912c6..ba19762 100644
--- a/arch/arm/mach-axxia/axxia.c
+++ b/arch/arm/mach-axxia/axxia.c
@@ -90,10 +90,8 @@ static void set_l3_pstate(u32 newstate)
 void
 flush_l3(void)
 {
-	/* Switch to SFONLY to flush */
-	set_l3_pstate(1);
-	/* ...and then back up again */
-	set_l3_pstate(3);
+	/* Shutdown to flush */
+	set_l3_pstate(0);
 }
 
 static struct map_desc axxia_static_mappings[] __initdata = {
-- 
1.7.5.4

